bundle/Resources/public/js/views/plugins/cof-createcontent-universaldiscoveryserviceplugin.js

```diff
@@ -297,6 +298,71 @@ YUI.add('cof-createcontent-universaldiscoveryserviceplugin', function (Y) {
         _deleteContent: function (event) {
             event.content.delete({api: this.get('host').get('capi')}, function () {});
         },
+
+        /**
+         * changeLanguage event handler. It opens languageSelectionBox for selecting
+         * language of created content
+         *
+         * @method _selectLanguage
+         * @param e {Object} event facade
+         */
+        _selectLanguage: function (e) {
+            e.preventDefault();
+            this.get('host').get('app').fire('languageSelect', {
+                config: {
+                    title: Y.eZ.trans('change.language.to', {}, 'contentedit'),
+                    languageSelectedHandler: Y.bind(this._setLanguage, this, e.target, e.fields),
+                    cancelLanguageSelectionHandler: null,
+                    canBaseTranslation: false,
+                    translationMode: true,
+                    referenceLanguageList: [this.get('host').get('languageCode')]
+                },
+            });
+        },
+
+       /**
+         * Sets language of created content to one given in event facade. After that notification is fired.
+         *
+         * @method _setLanguage
+         * @private
+         * @param {Y.eZ.View} view the view which triggered language selection box
+         * @param {Object} fields
+         * @param {EventFacade} e
+         * @param {String} e.selectedLanguageCode language code to which created content will be switched
+         */
+       _setLanguage: function (view, fields, e) {
+            var version = this.get('host').get('version'),
+                selectedLanguageCode = e.selectedLanguageCode,
+                formFields = {};
+
+            Y.Object.each(fields, function (field) {
+                formFields[field.fieldDefinitionIdentifier] = {
+                    fieldDefinitionIdentifier: field.fieldDefinitionIdentifier,
+                    fieldValue: field.fieldValue,
+                };
+            });
+
+            version.setFieldsIn(formFields, selectedLanguageCode);
+            this.get('host').set('languageCode', selectedLanguageCode);
+
+            view.setAttrs({
+                languageCode: selectedLanguageCode,
+                version: version,
+            });
+
+            this.fire('notify', {
```

Comment:
As far as I know, you cannot `fire` from the plugin. You have to fire on `this.get('host')`.